bitcoins/src/enc/bases.rs

```diff
@@ -1,117 +1,40 @@
-    #[error("Non-standard ScriptPubkey type")]
-    UnknownScriptType,
-
-    /// Bech32 HRP does not match the current network.
-    #[error("Bech32 HRP does not match. \nGot {:?} expected {:?} Hint: Is this address for another network?", got, expected)]
-    WrongHRP {
-        /// The actual HRP.
-        got: String,
-        /// The expected HRP.
-        expected: String,
-    },
-
-    /// Base58Check version does not match the current network
-    #[error("Base58Check version does not match. \nGot {:?} expected {:?} Hint: Is this address for another network?", got, expected)]
-    WrongVersion {
-        /// The actual version byte.
-        got: u8,
-        /// The expected version byte.
-        expected: u8,
-    },
-
-    /// Bubbled up error from base58check library
-    #[error("FromBase58CheckError: {:?}", .0)]
-    B58Error(FromBase58CheckError),
-
-    /// Bubbled up error from bech32 library
-    #[error("BechError: {:?}", .0)]
-    BechError(#[from] BechError),
-
-    /// Op Return ScriptPubkey was passed to encoder
-    #[error("Can't encode op return scripts as addresses")]
-    NullDataScript,
-}
-
-/// Impl explicitly because FromBase58CheckError doesn't implement the std error format
-impl From<FromBase58CheckError> for EncodingError {
-    fn from(e: FromBase58CheckError) -> Self {
-        EncodingError::B58Error(e)
-    }
-}
-
-/// A simple result type alias
-pub type EncodingResult<T> = Result<T, EncodingError>;
-
 /// Encode a byte vector to bech32. This function expects `v` to be a witness program, and will
 /// return an `UnknownScriptType` if it does not meet the witness program format.
-pub fn encode_bech32(hrp: &str, v: &[u8]) -> EncodingResult<String> {
+pub fn bitcoin_encode_bech32(hrp: &str, v: &[u8]) -> EncodingResult<String> {
     if v.len() < 2 {
         return Err(BechError::InvalidLength.into());
     }
+
     let (version_and_len, payload) = v.split_at(2);
     if version_and_len[0] > 16 || version_and_len[1] as usize != payload.len() {
         return Err(EncodingError::UnknownScriptType);
     };
-    let mut v = vec![u5::try_from_u8(version_and_len[0])?];
-    v.extend(&payload.to_base32());
-    b32_encode(hrp, &v).map_err(|v| v.into())
+
+    encode_bech32(hrp, version_and_len[0], &payload)
 }
 
 /// Decode a witness program from a bech32 string. Caller specifies an expected HRP. If a
 /// different HRP is found, returns `WrongHRP`.
-pub fn decode_bech32(expected_hrp: &str, s: &str) -> EncodingResult<Vec<u8>> {
-    let (hrp, data) = b32_decode(&s)?;
-    if hrp != expected_hrp {
-        return Err(EncodingError::WrongHRP {
-            got: hrp,
-            expected: expected_hrp.to_owned(),
-        });
-    }
-
-    // Extract the witness version and payload
-    let (v, p) = data.split_at(1);
-    let payload = Vec::from_base32(&p)?;
+pub fn bitcoin_decode_bech32(expected_hrp: &str, s: &str) -> EncodingResult<Vec<u8>> {
```

Comment:
I renamed the function to `bitcoin_decode_bech32` but maybe that is redundant since its in the `bitcoins` crate.

Reply:
seems redundant to me :thinking:

Reply:
Reverted this change. The consumer of the library will have to import the functions named differently if they want to encode/decode multiple address types in the same application, which seems like a more niche usecase.